superset/data/__init__.py

```diff
@@ -71,7 +70,7 @@ def load_energy():
     if not tbl:
         tbl = TBL(table_name=tbl_name)
     tbl.description = "Energy consumption"
-    tbl.database = get_or_create_main_db()
+    tbl.database = sm.get_or_create_main_db()
```

Comment:
Note this somewhat pre-dates this PR, but I'm not certain that `get_or_create_main_db` is even security related and thus having the security manager get or create the `main` database seems somewhat non-intuitive to me.

Reply:
Done

Reply:
Shouldn't this PR address this specific issue, especially as from this PR it would now indicate that the `get_or_create_main_db()` is tightly coupled with the security manager.

Reply:
My main concern is that moving this method will also make me move other things and make this PR even bigger and since it was already in security.py, it is not too far fetched to leave it there for now.

Reply:
Done!